src/client/hud.cpp

```diff
@@ -567,7 +652,21 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 				client->getMinimap()->drawMinimap(rect);
 				break; }
 			case HUD_ELEM_HOTBAR: {
-				drawHotbar(pos, e->offset, e->dir, e->align);
+				// Handle splitting caused by hud_hotbar_max_width
```

Comment:
Could you please move this code to a separate function? Also: `g_touchcontrols->resetHotbarRects()` does not need to be called twice.